cmake/FindRdKafka.cmake

```diff
@@ -8,48 +13,65 @@ else()
     set(RDKAFKA_SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX})
 endif()
 
-find_path(RDKAFKA_ROOT_DIR
-    NAMES include/librdkafka/rdkafka.h
-)
+set(RdKafka_LIBNAME ${RDKAFKA_PREFIX}rdkafka${RDKAFKA_SUFFIX})
 
-find_path(RDKAFKA_INCLUDE_DIR
+find_path(RdKafka_INCLUDE_DIR
     NAMES librdkafka/rdkafka.h
-    HINTS ${RDKAFKA_ROOT_DIR}/include
+    HINTS ${RdKafka_ROOT}/include
+)
+
+find_path(RdKafka_LIBRARY_DIR
+    NAMES ${RdKafka_LIBNAME} rdkafka
+    HINTS ${RdKafka_ROOT}/lib ${RdKafka_ROOT}/lib64
+)
+
+find_library(RdKafka_LIBRARY_PATH
+    NAMES ${RdKafka_LIBNAME} rdkafka
+    HINTS ${RdKafka_LIBRARY_DIR}
 )
 
 # Check lib paths
 if (CPPKAFKA_CMAKE_VERBOSE)
     get_property(FIND_LIBRARY_32 GLOBAL PROPERTY FIND_LIBRARY_USE_LIB32_PATHS)
     get_property(FIND_LIBRARY_64 GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS)
-    MESSAGE(STATUS "RDKAFKA search 32-bit library paths: ${FIND_LIBRARY_32}")
-    MESSAGE(STATUS "RDKAFKA search 64-bit library paths: ${FIND_LIBRARY_64}")
+    message(STATUS "RDKAFKA search 32-bit library paths: ${FIND_LIBRARY_32}")
+    message(STATUS "RDKAFKA search 64-bit library paths: ${FIND_LIBRARY_64}")
+    message(STATUS "RdKafka_ROOT = ${RdKafka_ROOT}")
+    message(STATUS "RdKafka_INCLUDE_DIR = ${RdKafka_INCLUDE_DIR}")
+    message(STATUS "RdKafka_LIBNAME = ${RdKafka_LIBNAME}")
+    message(STATUS "RdKafka_LIBRARY_PATH = ${RdKafka_LIBRARY_PATH}")
+    message(STATUS "RdKafka_LIBRARY_DIR = ${RdKafka_LIBRARY_DIR}")
 endif()
 
-find_library(RDKAFKA_LIBRARY
-    NAMES ${RDKAFKA_PREFIX}rdkafka${RDKAFKA_SUFFIX} rdkafka
-    HINTS ${RDKAFKA_ROOT_DIR}/lib
-)
-
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(RDKAFKA DEFAULT_MSG
-    RDKAFKA_LIBRARY
-    RDKAFKA_INCLUDE_DIR
+    RdKafka_LIBNAME
+    RdKafka_LIBRARY_DIR
+    RdKafka_LIBRARY_PATH
+    RdKafka_INCLUDE_DIR
 )
 
-set(CONTENTS "#include <librdkafka/rdkafka.h>\n #if RD_KAFKA_VERSION >= ${RDKAFKA_MIN_VERSION}\n int main() { }\n #endif")
+set(CONTENTS "#include <librdkafka/rdkafka.h>\n #if RD_KAFKA_VERSION >= ${RDKAFKA_MIN_VERSION_HEX}\n int main() { }\n #endif")
 set(FILE_NAME ${CMAKE_CURRENT_BINARY_DIR}/rdkafka_version_test.c)
 file(WRITE ${FILE_NAME} ${CONTENTS})
 
-try_compile(HAVE_VALID_KAFKA_VERSION ${CMAKE_CURRENT_BINARY_DIR}
+try_compile(RdKafka_FOUND ${CMAKE_CURRENT_BINARY_DIR}
             SOURCES ${FILE_NAME}
-            CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${RDKAFKA_INCLUDE_DIR}")
+            CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${RdKafka_INCLUDE_DIR}")
 
-if (HAVE_VALID_KAFKA_VERSION)
+if (RdKafka_FOUND)
+    if (CPPKAFKA_RDKAFKA_STATIC_LIB)
+        set(RdKafka_DEPENDENCIES ${RdKafka_LIBNAME} pthread rt ssl crypto dl z)
+    else()
+        set(RdKafka_DEPENDENCIES ${RdKafka_LIBNAME} pthread)
+    endif()
+    include_directories(SYSTEM ${RdKafka_INCLUDE_DIR})
+    link_directories(${RdKafka_LIBRARY_DIR})
```

Comment:
Hmmm but this script is used by CppKafka and also by any 3rd party library linking to CppKafka and using the CppKafkaConfig.cmake. So which target are we adding the include directories for? You can't say "target_include_directories(RdKafka...)` since we're not building this target, we are just importing it. Maybe I can do `add_library(RdKafka IMPORTED)` and then set the properties on the imported target. This is not something I have tried....will try and let you know.

Reply:
We've packaged a bunch of libraries at work using conan + CMake and we normally have the `Find*.cmake` script create an imported target and add all dependencies in there. This works great and lets you have all dependencies defined per target rather than having to remember to link with your dependencies' dependencies. We do something like

```cmake
add_library(foo INTERFACE IMPORTED)
set_target_properties(foo PROPERTIES
    INTERFACE_INCLUDE_DIRECTORIES ${FOO_INCLUDE_DIRS}
    INTERFACE_LINK_LIBRARIES "${FOO_LIBRARY};dep1;dep2;..."
)
```

Then you just import the library and link to it, ignoring dependencies (e.g. via `find_package` and `target_link_libraries`).

Reply:
Yeah this is what I just did. See my commit.